src/lib.rs

```diff
@@ -145,13 +149,24 @@ pub mod pallet {
             let vm =
                 Mvm::new(storage, balance.clone()).map_err(|_err| Error::<T>::ExecuteFailed)?;
 
-            let transaction = ScriptTransaction::try_from(transaction_bc.as_slice())
+            let tx = ScriptTransaction::try_from(transaction_bc.as_slice())
                 .map_err(|_| Error::<T>::ExecuteFailed)?;
 
+            let signer_count = verify_script_integrity_and_check_signers(&tx.bytecode)
+                .map_err(Error::<T>::from)?;
+
+            let mut signature_handler = ScriptSignatureHandler::<T>::new(&tx.args, signer_count)?;
+            signature_handler.sign_script(&who)?;
+
+            if !signature_handler.all_signers_approved() {
+                // TODO(eiger): Use another approach for the multi-signature feature.
+                return Err(Error::<T>::ScriptSignatureFailure.into());
+            }
+
```

Comment:
```rust
            if !signature_handler.all_signers_approved() {
                // TODO(eiger): Use another approach for the multi-signature feature.
                return Err(Error::<T>::ScriptSignatureFailure.into());
            }
```
A note here:
We can easily extend this for the multi-signature feature.

After one signer (user) signs the script which requires multiple signers - if we don't have all required signatures, we would store the whole `signature_handler` for that script in a special storage:
```rust
HashMap<HashOf(ScriptTransaction), ScriptSignatureHandler>
```
Then, the next user (missing signer) would need to sign the same script - and we would first check if we have the `ScriptSignatureHandler` for that `ScriptTransaction` in the storage - if yes, we would append the new signature, and if all signatures are there - we would execute the script finally - a script with multiple signatures 😤 


The only question is: 
How long should we keep a partially signed `ScriptTransaction` in the storage? 🤔